src/data/nat/with_bot.lean

```diff
@@ -13,24 +13,39 @@ Lemmas about the type of natural numbers with a bottom element adjoined.
 -/
 namespace nat
 
-lemma with_bot.add_eq_zero_iff : ∀ {n m : with_bot ℕ}, n + m = 0 ↔ n = 0 ∧ m = 0
-| none     m        := iff_of_false dec_trivial (λ h, absurd h.1 dec_trivial)
-| n        none     := iff_of_false (by cases n; exact dec_trivial)
-  (λ h, absurd h.2 dec_trivial)
-| (some n) (some m) := show (n + m : with_bot ℕ) = (0 : ℕ) ↔ (n : with_bot ℕ) = (0 : ℕ) ∧
-    (m : with_bot ℕ) = (0 : ℕ),
-  by rw [← with_bot.coe_add, with_bot.coe_eq_coe, with_bot.coe_eq_coe,
-    with_bot.coe_eq_coe, add_eq_zero_iff' (nat.zero_le _) (nat.zero_le _)]
-
-lemma with_bot.add_eq_one_iff : ∀ {n m : with_bot ℕ}, n + m = 1 ↔ (n = 0 ∧ m = 1) ∨ (n = 1 ∧ m = 0)
-| none     none     := dec_trivial
-| none     (some m) := dec_trivial
-| (some n) none     := iff_of_false dec_trivial (λ h, h.elim (λ h, absurd h.2 dec_trivial)
-  (λ h, absurd h.2 dec_trivial))
-| (some n) (some 0) := by erw [with_bot.coe_eq_coe, with_bot.coe_eq_coe, with_bot.coe_eq_coe,
-    with_bot.coe_eq_coe]; simp
-| (some n) (some (m + 1)) := by erw [with_bot.coe_eq_coe, with_bot.coe_eq_coe, with_bot.coe_eq_coe,
-    with_bot.coe_eq_coe, with_bot.coe_eq_coe]; simp [nat.add_succ, nat.succ_inj', nat.succ_ne_zero]
+lemma with_bot.add_eq_zero_iff {n m : with_bot ℕ} : n + m = 0 ↔ n = 0 ∧ m = 0 :=
```

Comment:
You might consider adding `namespace with_bot` at the top of the file to shorten all of the lemma names.